server/src/main/java/org/exoplatform/chat/services/mongodb/UserServiceImpl.java

```diff
@@ -493,6 +504,121 @@ private SpaceBean getSpace(String roomId, String dbName)
+    orList.add(un);
+    orList.add(fn);
+    BasicDBObject filterQuery = new BasicDBObject("$or", orList);
+
+    // Search in team members first
+    if (roomId != null && !"".equals(roomId)) {
+      // Search in room first
+      //removing "space-" prefix
+      if (roomId.indexOf(ChatService.SPACE_PREFIX)==0) {
+        roomId = roomId.substring(ChatService.SPACE_PREFIX.length());
+      }
+      //removing "team-" prefix
+      if (roomId.indexOf(ChatService.TEAM_PREFIX)==0) {
+        roomId = roomId.substring(ChatService.TEAM_PREFIX.length());
+      }
+
+      BasicDBObject spaces = new BasicDBObject("spaces", roomId);
+      BasicDBObject teams = new BasicDBObject("teams", roomId);
+      orList = new ArrayList<BasicDBObject>();
+      orList.add(spaces);
+      orList.add(teams);
+      BasicDBObject teamQuery = new BasicDBObject("$or", orList);
+
+      BasicDBObject query = new BasicDBObject("$and", Arrays.asList(teamQuery, filterQuery));
+      searchByQuery(users, dbName, query, limit);
+    }
+
+    int remain = limit - users.size();
+
+    // Then search in connection
+    if (remain > 0 && currentUserName != null && !"".equals(currentUserName)) {
+      // We can not inject IdentityManager to this service here because it threw NPE,
+      // maybe Guice container only return this service instance when portal container started
+      IdentityManager identityManager = GuiceManager.getInstance().getInstance(IdentityManager.class);
+      if (identityManager != null) {
+        Identity identity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, currentUserName, false);
+        if (identity != null && !identity.isDeleted()) {
+          try {
+            ListAccess<Identity> connections = identityManager.getConnectionsWithListAccess(identity);
+            if (connections != null && connections.getSize() > 0) {
+              int start = 0;
+              int size = connections.getSize();
+              while (start < size && users.size() < limit) {
+                Identity[] ids = connections.load(0, limit);
+                for (Identity id : ids) {
+                  Profile profile = identityManager.getProfile(id);
+                  if (!users.containsKey(id.getRemoteId()) && isMatch(originalFilter, profile)) {
+                    UserBean userBean = new UserBean();
+                    userBean.setName(id.getRemoteId());
+                    userBean.setEmail(profile.getEmail());
+                    userBean.setFullname(profile.getFullName());
+                    userBean.setStatus(UserService.STATUS_AVAILABLE);
+
+                    users.put(id.getRemoteId(), userBean);
+                  }
+                  if (users.size() >= limit) break;
+                }
+                start += limit;
+              }
+            }
+          } catch (Exception ex) {
+            LOG.info(ex.getMessage());
+          }
+          //
+          remain = limit - users.size();
+        }
+      }
+    }
```

Comment:
This code block of finding connections by user info should be moved to Social service, even we may need to introduce a new method.

It helps support another social RDMBS implementation add-on works well later.